x/app-router/types/parser_test.go

```diff
@@ -25,45 +35,23 @@ func TestParseReceiverDataNoTransfer(t *testing.T) {
 }
 
 func TestParseReceiverDataErrors(t *testing.T) {
-	testCases := []struct {
-		name          string
-		data          string
-		errStartsWith string
-	}{
-		{
-			"unparsable transfer field",
-			"",
-			"unparsable receiver",
-		},
-		{
-			"unparsable transfer field",
-			"abc:def:",
-			"unparsable receiver",
-		},
-		{
-			"missing pipe",
-			"transfer/channel-0:cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
-			"formatting incorrect",
-		},
-		{
-			"invalid this chain address",
-			"somm16plylpsgxechajltx9yeseqexzdzut9g8vla4k|transfer/channel-0:cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
-			"decoding bech32 failed",
-		},
-		{
-			"missing slash",
-			"cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k|transfer\\channel-0:cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
-			"formatting incorrect",
-		},
-	}
+	// empty transfer field
+	pt, err := ParseReceiverData("")
+	require.NoError(t, err)
+	require.False(t, pt.ShouldLiquidStake)
 
-	for _, tc := range testCases {
-		tc := tc
+	// invalid string
+	pt, err = ParseReceiverData("abc:def:")
+	require.NoError(t, err)
+	require.False(t, pt.ShouldLiquidStake)
+
+	// invalid function
+	pt, err = ParseReceiverData("stride10d07y265gmmuvt4z0w9aw880jnsr700jefnezl|stakeibc/xxx")
+	require.NoError(t, err)
+	require.False(t, pt.ShouldLiquidStake)
 
-		t.Run(tc.name, func(t *testing.T) {
-			_, err := ParseReceiverData(tc.data)
-			require.Error(t, err)
-			require.Equal(t, err.Error()[:len(tc.errStartsWith)], tc.errStartsWith)
-		})
-	}
+	// invalid address
+	pt, err = ParseReceiverData("xxx|stakeibc/liquidstake")
+	require.Error(t, err)
```

Comment:
Can you use `EqualError` here?

Reply:
Done https://github.com/Stride-Labs/stride/pull/432/commits/d17ffb6028a32eba576bf75d10c226dfc539278e